gems/pending/VMwareWebService/MiqVimVm.rb

```diff
@@ -850,25 +850,18 @@ def removeDiskByFile(backingFile, deleteBacking = false)
   def getScsiCandU
     devs = getProp("config.hardware")["config"]["hardware"]["device"]
     ctrlrHash = {}
-    devs.each do |dev|
-      next unless (key = dev["key"])
-      next unless dev["deviceInfo"]
-      next unless (label = dev["deviceInfo"]["label"])
-      ctrlrHash[key] = 0 if label =~ /^SCSI\s[Cc]ontroller\s.*$/
-    end
+
+    scsi_dev = devs.detect {|x| x.xsiType == 'ParaVirtualSCSIController' } || {}
```

Comment:
You'll need to check for other SCSI controller types as well, see "Extended by" here https://www.vmware.com/support/developer/vc-sdk/visdk41pubs/ApiReference/vim.vm.device.VirtualSCSIController.html

We do something similar here https://github.com/ManageIQ/manageiq/blob/master/gems/pending/VMwareWebService/MiqVimVm.rb#L535-L541